packages/ember-metal/lib/meta.js

```diff
@@ -326,7 +327,7 @@ export class Meta {
     let descriptor = lookupDescriptor(obj, key);
     let isMandatorySetter = descriptor && descriptor.set && descriptor.set.isMandatorySetter;
 
-    if (isMandatorySetter) {
+    if (isMandatorySetter === true) {
```

Comment:
This looks like overkill to me, because if you use `!==` in the individual `&&` cases above, then the compiler knows statically that `isMandatorySetter` is Boolean.